### ISSUE TITLE
"remove multiple PIO Cores from a system" – how?

### ISSUE THREAD

**Issue by MacDada** [platformio/platformio-core#4664]
`pio run` gives me:

```
Obsolete PIO Core v6.1.7 is used (previous was 6.1.8a1)
Please remove multiple PIO Cores from a system:
https://docs.platformio.org/en/latest/core/installation/troubleshooting.html
```

So I follow the advice from the link, uninstalled PIO, installed it again -> same problem.

I've been away from my project for awhile, came back to it just now, this is a new issue for me, didn't happen before.

MacOS Monterey, 12.6.6, 21G646, Intel, CLion, PlatformIO CLI

```
➜  DnWiFiDoorLock git:(master) ✗ pio run
***********************************************************************************************************************************************************************************
Obsolete PIO Core v6.1.7 is used (previous was 6.1.8a1)
Please remove multiple PIO Cores from a system:
https://docs.platformio.org/en/latest/core/installation/troubleshooting.html
***********************************************************************************************************************************************************************************
Processing native (platform: native)
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
^CBuild interrupted.
Error: Aborted by user

➜  DnWiFiDoorLock git:(master) ✗ pip uninstall platformio
WARNING: Skipping platformio as it is not installed.

➜  DnWiFiDoorLock git:(master) ✗ python -m pip uninstall platformio
zsh: command not found: python

➜  DnWiFiDoorLock git:(master) ✗ python3 -m pip uninstall platformio
WARNING: Skipping platformio as it is not installed.

➜  DnWiFiDoorLock git:(master) ✗ brew uninstall platformio
Uninstalling /usr/local/Cellar/platformio/6.1.7... (1,861 files, 18.2MB)

➜  DnWiFiDoorLock git:(master) ✗ pio
zsh: command not found: pio

➜  DnWiFiDoorLock git:(master) ✗ brew install platformio
==> Fetching platformio
==> Downloading https://ghcr.io/v2/homebrew/core/platformio/manifests/6.1.7-1
Already downloaded: /Users/dada/Library/Caches/Homebrew/downloads/00335ea4f695dc5795b2a6f7be638c7149daac647fa91527213aa15a20cf8d36--platformio-6.1.7-1.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/platformio/blobs/sha256:92a98652f70fd418fa54eb19251d24a38136fbbfdb15e7b65fb7404fa01212e9
Already downloaded: /Users/dada/Library/Caches/Homebrew/downloads/3c3a3b57f802e4e385af2770e17e39ee63756abac8028dab1bc40b196b13ea3b--platformio--6.1.7.monterey.bottle.1.tar.gz
==> Pouring platformio--6.1.7.monterey.bottle.1.tar.gz
🍺  /usr/local/Cellar/platformio/6.1.7: 1,623 files, 15.5MB
==> Running `brew cleanup platformio`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).

➜  DnWiFiDoorLock git:(master) ✗ pio run
***********************************************************************************************************************************************************************************
Obsolete PIO Core v6.1.7 is used (previous was 6.1.8a1)
Please remove multiple PIO Cores from a system:
https://docs.platformio.org/en/latest/core/installation/troubleshooting.html
***********************************************************************************************************************************************************************************
Processing native (platform: native)
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
^CBuild interrupted.
Error: Aborted by user

➜  DnWiFiDoorLock git:(master) ✗ which pio
/usr/local/bin/pio

➜  DnWiFiDoorLock git:(master) ✗ ls -alh /usr/local/bin/pio
lrwxr-xr-x  1 dada  admin    34B Jun 16 02:17 /usr/local/bin/pio -> ../Cellar/platformio/6.1.7/bin/pio

➜  DnWiFiDoorLock git:(master) ✗ platformio --version
PlatformIO Core, version 6.1.7

➜  DnWiFiDoorLock git:(master) ✗ which platformio
/usr/local/bin/platformio

➜  DnWiFiDoorLock git:(master) ✗ ls -alh /usr/local/bin/platformio
lrwxr-xr-x  1 dada  admin    41B Jun 16 02:17 /usr/local/bin/platformio -> ../Cellar/platformio/6.1.7/bin/platformio
```

**Comment by ivankravets** [comment]
Just remove `%HOME_DIR%/.platformio/appstate.json`.

**Comment by MacDada** [comment]
> Just remove `%HOME_DIR%/.platformio/appstate.json`.

It worked, thx!

> [ivankravets](https://github.com/ivankravets) closed this as [completed](https://github.com/platformio/platformio-core/issues?q=is%3Aissue+is%3Aclosed+archived%3Afalse+reason%3Acompleted) [24 minutes ago](https://github.com/platformio/platformio-core/issues/4664#event-9551499537)

Isn't this some kind of a bug? If not, what did _I_ do wrong to cause it? Shouldn't the solution be listed in https://docs.platformio.org/en/latest/core/installation/troubleshooting.html ?

**Comment by universalturingmachine** [comment]
I got the same problem. Removing the %HOME_DIR%/.platformio/appstate.json file was a temporary solution. The problem happened because of this.

I have Platform IO VS Code IDE and Platform IO CLI. The IDE has the version 6.1.9. The CLI has the version 6.1.1. If I compile from the IDE, it records the version in the appstate.json file. Then if I compile from the CLI, it checks the version and finds that 6.1.1 is obsolete. That's why that line is printed. This says that there are 2 different versions.

I solved it by running pio upgrade from CLI. Now both CLI and the IDE are in 6.1.9.

Ideally IDE and CLI should have the same environment to run. But that is not so for me. Trying to fix it.
